sympy/physics/mechanics/abstract_body.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
```

Comment:
Named the file `abstract_body.py`, because we are currently using separate files for all bodies and this way we can avoid circular import errors. Another option would be to put the bodies in the same `bodies.py` file, but not sure if this is allowed with respect to backward compatibility. Besides of the fact that the file will be long.

P.S. A better name is also welcome.